include/xcp-ng/generic/io.h

```diff
@@ -51,12 +53,20 @@ XcpError xcp_fd_wait_read (int fd, void *buf, size_t count, int timeout);
 // - Broken socket.
 XcpError xcp_fd_read_all (int fd, void *buf, size_t count, int timeout, size_t *offset);
 
+// -----------------------------------------------------------------------------
+
 // Write. (/!\ Do not use (bypass) the O_NONBLOCK flag. /!\)
 XcpError xcp_fd_write (int fd, const void *buf, size_t count);
 
 // Wait and write `count` bytes.
 XcpError xcp_fd_write_all (int fd, const void *buf, size_t count, size_t *offset);
 
+// -----------------------------------------------------------------------------
+
+XcpError xcp_fd_pread (int fd, void *buf, size_t count, off_t offset);
```

Comment:
Should we also make a `xcp_fd_pread_all`? :thinking:

Reply:
What about `pwrite` and `pwrite_all`

Reply:
`xcp_fd_pread_all`== `xcp_fd_pread`. Read the `pread` documentation. :wink:
I don't need to create a `pwrite` function for the moment. Maybe for a future project.